postgresqlPackages.timescaledb: 2.14.2 -> 2.17.2; adopt, nixfmt; postgresqlPackages.timescaledb_toolkit: 1.18.0 -> 1.19.0

kirillrdy opened the pull request:
Changes:
-  https://github.com/timescale/timescaledb/releases/tag/2.17.2
-  https://github.com/timescale/timescaledb/releases/tag/2.17.1
 - https://github.com/timescale/timescaledb/releases/tag/2.17.0
 - https://github.com/timescale/timescaledb/releases/tag/2.16.1
 - https://github.com/timescale/timescaledb/releases/tag/2.16.0
 - https://github.com/timescale/timescaledb/releases/tag/2.15.3
 - https://github.com/timescale/timescaledb/releases/tag/2.15.2
 - https://github.com/timescale/timescaledb/releases/tag/2.15.1
 - https://github.com/timescale/timescaledb/releases/tag/2.15.0
 - https://github.com/timescale/timescaledb-toolkit/releases/tag/1.19.0

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

kirillrdy committed:
postgresqlPackages.timescaledb: 2.14.2 -> 2.17.2

Changes:
 - https://github.com/timescale/timescaledb/releases/tag/2.17.2
 - https://github.com/timescale/timescaledb/releases/tag/2.17.1
 - https://github.com/timescale/timescaledb/releases/tag/2.17.0
 - https://github.com/timescale/timescaledb/releases/tag/2.16.1
 - https://github.com/timescale/timescaledb/releases/tag/2.16.0
 - https://github.com/timescale/timescaledb/releases/tag/2.15.3
 - https://github.com/timescale/timescaledb/releases/tag/2.15.2
 - https://github.com/timescale/timescaledb/releases/tag/2.15.1
 - https://github.com/timescale/timescaledb/releases/tag/2.15.0

squash

kirillrdy committed:
postgresqlPackages.timescaledb: add kirillrdy to maintainers

wolfgangwalther reviewed (changes requested):
Could you squash the two update commits + the "only supports postgresql 14" together in one commit and keep the maintainer + nixfmt commits separate?

kirillrdy commented:
@wolfgangwalther going to draft until after 24.11 branch off

kirillrdy committed:
postgresqlPackages.timescaledb: nixfmt

kirillrdy committed:
nixos/doc/rl-2505: mention timescaledb

kirillrdy commented on pkgs/servers/sql/postgresql/ext/timescaledb.nix:
```diff
@@ -32,16 +51,18 @@ stdenv.mkDerivation rec {
+    broken = versionOlder postgresql.version "14" ||
       # timescaledb supports PostgreSQL 17 from 2.17.0 on:
       # https://github.com/timescale/timescaledb/releases/tag/2.17.0
       # We can't upgrade to it, yet, because this would imply dropping support for
```
done

kirillrdy commented on pkgs/servers/sql/postgresql/ext/timescaledb.nix:
```diff
@@ -1,20 +1,39 @@
+  openssl,
+  libkrb5,
+  nixosTests,
+  enableUnfree ? true,
```
done

kirillrdy commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 348223`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>postgresql13JitPackages.timescaledb</li>
    <li>postgresql13JitPackages.timescaledb-apache</li>
    <li>postgresql13Packages.timescaledb</li>
    <li>postgresql13Packages.timescaledb-apache</li>
  </ul>
</details>
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>postgresql14JitPackages.timescaledb</li>
    <li>postgresql14JitPackages.timescaledb-apache</li>
    <li>postgresql14Packages.timescaledb</li>
    <li>postgresql14Packages.timescaledb-apache</li>
    <li>postgresql15JitPackages.timescaledb</li>
    <li>postgresql15JitPackages.timescaledb-apache</li>
    <li>postgresql15Packages.timescaledb</li>
    <li>postgresql15Packages.timescaledb-apache</li>
    <li>postgresql16JitPackages.timescaledb</li>
    <li>postgresql16JitPackages.timescaledb-apache</li>
    <li>postgresql16Packages.timescaledb</li>
    <li>postgresql16Packages.timescaledb-apache</li>
    <li>postgresql17JitPackages.timescaledb</li>
    <li>postgresql17JitPackages.timescaledb-apache</li>
    <li>postgresql17Packages.timescaledb</li>
    <li>postgresql17Packages.timescaledb-apache</li>
  </ul>
</details>

wolfgangwalther reviewed (changes requested):
Could you squash the last commit "mention timescaledb" into the second commit "2.14.2 -> 2.17.1"?

Those changes really belong together, I think.

kirillrdy commented:
> > There is now also an update to `timescaledb_toolkit` available, which adds support for PG17. I suggest you bump that here, too.
> 
> Did you have a look at that, too?
> 
> Without bumping timescaledb_toolkit we can't test `nixosTests.postgresql.timescaledb`, I think.

I can have a look into `timescaledb_toolkit`, `nixosTests.postgresql.timescaledb` builds, but master is still on 16.4

philon123 commented:
@kirillrdy you're awesome, thanks! Can I help with this somehow? I could maybe confirm it working on x86_64 with a real world use case. Not sure if I can contribute to the actual PR though.

Would it be fine to update to the newest version, v2.17.2?

Any idea if this will make NixOS 24.11?

kirillrdy commented:
> @kirillrdy you're awesome, thanks! Can I help with this somehow? I could maybe confirm it working on x86_64 with a real world use case. Not sure if I can contribute to the actual PR though.
> 
> Would it be fine to update to the newest version, v2.17.2?
> 
> Any idea if this will make NixOS 24.11?

any testing is much appreciated, I only looked at whats in `nixosTests.timescaledb`

yes I will bump to  v2.17.2

no, this will not make to NixOS 24.11 due to breaking changes, but it doesn't mean it can not be installed on nixos 24.11

kirillrdy committed:
postgresqlPackages.timescaledb_toolkit: 1.18.0 -> 1.19.0

Release notes: https://github.com/timescale/timescaledb-toolkit/releases/tag/1.19.0

kirillrdy commented:
> `postgresql17Packages.timescaledb_toolkit` doesn't build for me with this PR:
> 
> ```
>        > error: unexpected argument '--pg17' found
>        >
>        >   tip: a similar argument exists: '--pg16'
> ```

yes, looks like we need to bump cargo-pgrx too

kirillrdy committed:
cargo-pgrx_0_12_6: init at 0_12_6

kirillrdy committed:
postgresqlPackages.timescaledb_toolkit: switch to cargo-pgrx_0_12_6

Ma27 commented on pkgs/development/tools/rust/cargo-pgrx/default.nix:
```diff
@@ -77,4 +77,11 @@ in
     cargoHash = "sha256-9XTIcpoCnROP63ZTDgMMMmj0kPggiTazKlKQfCgXKzk=";
   };
+
+  cargo-pgrx_0_12_6 = generic {
```
Can't we bump the alpha version above?

kirillrdy replied:
This breaks vector-rs I'll have a look into that

Ma27 replied:
Ah, bummer.
Nvm, you don't have to fix it up in this PR.

Ma27 replied:
> Should we remove it?

Go for it.
I wouldn't do it on stable, but that's not the case here.

>  I wonder whether we should take responsibility for this with the @NixOS/postgres team as well?

Sounds good. Do you want to file a PR for that?

Ma27 reviewed (commented):
Ah right, forgot to check the v17 variant of the toolkit. Good thing I got distracted from merging I guess ;-)

wolfgangwalther reviewed (approved):
Result of `nixpkgs-review pr 348223` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>postgresql13JitPackages.timescaledb</li>
    <li>postgresql13JitPackages.timescaledb-apache</li>
    <li>postgresql13Packages.timescaledb</li>
    <li>postgresql13Packages.timescaledb-apache</li>
  </ul>
</details>
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>27 packages built:</summary>
  <ul>
    <li>cargo-pgrx_0_12_6</li>
    <li>postgresql13JitPackages.timescaledb_toolkit</li>
    <li>postgresql13Packages.timescaledb_toolkit</li>
    <li>postgresql14JitPackages.timescaledb</li>
    <li>postgresql14JitPackages.timescaledb-apache</li>
    <li>postgresql14JitPackages.timescaledb_toolkit</li>
    <li>postgresql14Packages.timescaledb</li>
    <li>postgresql14Packages.timescaledb-apache</li>
    <li>postgresql14Packages.timescaledb_toolkit</li>
    <li>postgresql15JitPackages.timescaledb</li>
    <li>postgresql15JitPackages.timescaledb-apache</li>
    <li>postgresql15JitPackages.timescaledb_toolkit</li>
    <li>postgresql15Packages.timescaledb</li>
    <li>postgresql15Packages.timescaledb-apache</li>
    <li>postgresql15Packages.timescaledb_toolkit</li>
    <li>postgresql16JitPackages.timescaledb</li>
    <li>postgresql16JitPackages.timescaledb-apache</li>
    <li>postgresql16JitPackages.timescaledb_toolkit</li>
    <li>postgresql16Packages.timescaledb</li>
    <li>postgresql16Packages.timescaledb-apache</li>
    <li>postgresql16Packages.timescaledb_toolkit</li>
    <li>postgresql17JitPackages.timescaledb</li>
    <li>postgresql17JitPackages.timescaledb-apache</li>
    <li>postgresql17JitPackages.timescaledb_toolkit</li>
    <li>postgresql17Packages.timescaledb</li>
    <li>postgresql17Packages.timescaledb-apache</li>
    <li>postgresql17Packages.timescaledb_toolkit</li>
  </ul>
</details>

---

`nixosTests.postgresql.timescaledb` also built fine for all versions.
